src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementEventNormalizationService.cs

```diff
@@ -88,6 +90,13 @@ private async Task StartConsumer(ISourceBlock<EventData> producer, IAsyncCollect
                             (DateTime.UtcNow - evt.SystemProperties.EnqueuedTimeUtc).TotalMilliseconds);
 
                         var token = _converter.Convert(evt);

```

Comment:
I am wondering if we have an opportunity to do something more efficiently in the EventDataWithJsonBodyToJTokenConverter or prior based off the raw EventData.

The EventData contains the byte array for the body so we can just record the size of that for the *data*.  The question is do we include the size of the properties/system properties (if we want to which your current method does).  If the performance impact ends up being high enough I think it is a reasonable compromise to just meter the 

It also might be worth looking at the latest https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs SDK to see if there is a method for calculating the size of the event message built in.

The one benefit of your approach that I like is it applies no matter the source as long as we convert the payload to JSON I just have a feeling we will want to end up with something more specialized per converter implementation to get the necessary efficiency to sustain throughput.